docs/pull_request_template.md

```diff
@@ -0,0 +1,64 @@
+# Description
+
+Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
+
+You can refer to [Kubernete's community documentation] on writing good commit messages, which provides good tips and ideas.
+
+Some PRs addess specific issues. Please, refer to the [CONTRIBUTING] documentation for more
+information on how to link a PR to an existing issue.
+
+# What environments does this code impact?
+
+- [ ] Cloud
+- [ ] Onprem
```

Comment:
Onprem is a bit of a tough term. What do we mean by "onprem" if we also have "operator" here?
Also "operator" and "controller" could be confusing. Maybe we need some comments in the template that explain when these would need to be checked?

For example, do I check "Operator" only when I'm changing the agent service config controller or would I use it also when changing local auth (which is only used currently by the serice when deployed with the operator)?

Reply:
> For example, do I check "Operator" only when I'm changing the agent service config controller or would I use it also when changing local auth (which is only used currently by the service when deployed with the operator)?

You make a good point. I will remove `onprem` as I had added it before adding `Controller` and `Operator` but I will leave the comments/docs on what each part means for a follow-up PR so that we can start moving this one along.